File: backend/Models/ParkingLot.js
const mongoose = require('mongoose')

const ParkingLotSchema = new mongoose.Schema({
    name: {
        type: String,
        required: [true, 'Please provide a name']
    },
    email: {
        type: String,
        required: true,
        unique: true,
        match: [
            /^\w+([\.-]?\w+)*@\w+([\.-]?\w+)*(\.\w{2,3})+$/,
            'Please add a valid email'
        ],
    },
    phone: {
        type: Number
    },
    password: {
        type: String,

    },
    // googleId: {
    //     type: String
    // },
    // facebookId: {
    //     type: String
    // }
    address: {
        type: String,

    },
    latitude: {
        type: Number
    },
    longitude: {
        type: Number
    },
    image: {
        type:
            [
                {
                    type: String, default: null
                }
            ]
    },
    charges: {
        type: Number,

    },
    startingTime: {
        type: String,
        default: "00"
    },
    endingTime: {
        type: String,
        default: "23"
    },
    FourSpotsLeft: {
        type: Number,
        default: 0
    },
    TwoSpotsLeft: {
        type: Number,
        default: 0
    },
    MapApiData: {
        type: String
    },
    averageRating: {
        type: Number,
        min: [0, 'Rating must be at least 0'],
        max: [5, 'Rating must can not be more than 5'],
        default: 0
    },
    averageCount: {
        type: Number,
        default: 0
    }

})

module.exports = mongoose.model('ParkingLot', ParkingLotSchema);

File: frontend/park_kar/lib/models/parkingLot.dart
class ParkingModel {
  String id;
  String name;
  String email;
  int phone;
  String address;
  num latitude;
  num longitude;
  String image;
  num charges;
  String startingTime;
  String endingTime;
  int fourSpotsLeft;
  int twoSpotsLeft;
  String mapApiData;
  num averageRating;
  String distance;
  num distanceValue;
  String duration;

  ParkingModel(
      {this.id,
      this.name,
      this.email,
      this.phone,
      this.address,
      this.latitude,
      this.longitude,
      this.image,
      this.charges,
      this.startingTime,
      this.endingTime,
      this.fourSpotsLeft,
      this.twoSpotsLeft,
      this.mapApiData,
      this.averageRating});

  factory ParkingModel.fromJson(Map<String, dynamic> json) {
    return ParkingModel(
      id: json['_id'] as String,
      name: json['name'] as String,
      email: json['email'] as String,
      phone: json['phone'] as int,
      address: json['address'] as String,
      latitude: json['latitude'] as num,
      longitude: json['longitude'] as num,
      image: json['image'][0] as String,
      charges: json['charges'] as num,
      startingTime: json['startingTime'] as String,
      endingTime: json['endingTime'] as String,
      fourSpotsLeft: json['FourSpotsLeft'] as int,
      twoSpotsLeft: json['TwoSpotsLeft'] as int,
      mapApiData: json['mapApiData'] as String,
      averageRating: json['averageRating'] as num,
    );
  }
}
